src/main/java/Command.java

```diff
@@ -22,6 +25,22 @@ public String getName() {
         return name;
     }
 
+
+    protected Matcher parseUserInput(String regex, String data) {
```

Comment:
Can consider changing the variable name "regex" to something more easily understood.